test/access-control/int.spec.ts

```diff
@@ -522,6 +524,44 @@ describe('Access Control', () => {
 
       expect(docs).toHaveLength(1)
     })
+
+    it('should ignore false access on query constraint added by top collection level access control', async () => {
+      await payload.create({
+        collection: 'fields-and-top-access',
+        data: { secret: '123456' },
```

Comment:
This test is very hard to read what its expecting. Instead of querying for `12345` and `123456`, I set the value to `will-fail-access-read`, assign it to a variable like `hit` or `miss`, or use comments. I had to go back and check the access control `read` for this collection to figure this out.